src/library/type_context.cpp

```diff
@@ -3994,6 +3994,37 @@ static bool depends_on_mvar(expr const & e, buffer<expr> const & mvars) {
     return false;
 }
 
+buffer<bool> type_context_old::class_out_param_deps(expr const & cls) {
```

Comment:
```suggestion
void class_out_param_deps(expr const & cls_type, buffer<bool> & is_out_param) {
```
Where `cls_type` is the type of the class constant.  Both call sites already have that type.  Then this function wouldn't need to be in type_context either.